api/src/list/dao/list.dao.ts

```diff
@@ -18,47 +19,34 @@ export class ListDao {
     throw new BadRequestException('List is private');
   }
 
-  async getList(listId: string) {
+  async getList(listId: string, query?: QueryDto) {
     const list = await this.prisma.list.findUniqueOrThrow({
       where: { id: listId },
-      select: {
-        id: true,
-        name: true,
-        isPrivate: true,
-        creatorId: true,
-        createdAt: true,
-        updatedAt: true,
-        movie: true,
-        user: true,
+      include: {
+        movie: {
+          take: query?.per_page || 10,
+          skip: (query?.page_number || 0) * (query?.per_page || 0),
+        },
```

Comment:
I think this kind of breaks the work we did before. This is the `list` service so it should only take care of list stuff (in this case, it'd return info about the list like name, isPrivate, and posterId once we have it), The `movie` service is the one in charge of delivering movie info, and I think pagination should happen in `getMovies`.